### PR TITLE
Never run git-ignored hooks, regardless of permissions

### PR DESCRIPTION
**Pull request by codebling**

### CODE REVIEW

**Comment by codebling** [comment]
Fixes #30.

**Comment by tarmolov** [comment]
I didn't get an issue which you try to solve with this pr.

**Comment by codebling** [comment]
Consider the [test on line 121 of `run.test.js`](https://github.com/tarmolov/git-hooks-js/blob/master/tests/run.test.js#L121). 

    describe('do not run git-ignored scripts from hooks directory',
        it('should ignore file with wrong permissions in hooks directory',

What is this testing?  Files with wrong permissions will not be run, so this test is really a duplicate of the test [on line 46](https://github.com/tarmolov/git-hooks-js/blob/master/tests/run.test.js#L46), and so nothing related to git-ignored files is tested. 

In fact, git-ignored files will often still be executed, if they are using non-trivial paths or if there is more than one git-ignored file in the git repository. See #30. 

This PR ensures that git-ignored files are never executed, and that this functionality will also work on Windows.

File: lib/git-hooks.js [hunk 1]
```diff
@@ -33,8 +33,26 @@ var HOOKS = [
...
+        output = typeof output !== 'string' ? '' : output; // basic sanity check
+        var ignoredHookPaths = output.split(/\r?\n/); // split string into array of each ignored file
+
+        ignoredHookPaths = ignoredHookPaths.filter(function (ignoredHookPath) {
+            // remove empty strings from the results
+            return ignoredHookPath.length > 0;
+        });
+        var normalizePath = function (hookPath) {
+            // normalize paths
+            if (process.platform === 'win32') {
```
Comment (tarmolov) [inline lib/git-hooks.js]: Let's remove windows support in this pr. It's better to add it in a separate pr.

File: lib/git-hooks.js [hunk 2]
```diff
@@ -33,8 +33,26 @@ var HOOKS = [
 function excludeIgnoredPaths(hooks, callback) {
     exec('git check-ignore ' + hooks.join(' '), function (error, output) {
         // intentionally ignore errors
-        callback(hooks.filter(function (hookName) {
-            return output.indexOf(hookName) === -1;
+        output = typeof output !== 'string' ? '' : output; // basic sanity check
```
Comment (tarmolov) [inline lib/git-hooks.js]: Also please remove all inline comments such as `basic sanity check`. They don't make sense.

**Comment by codebling** [comment]
Feel free to make changes as you see fit. I have this repo set up on a Windows env and several tests fail without Windows support.

**Comment by tarmolov** [comment]
Fixed in #41

**Pull request closed without merging**